alica_engine/include/engine/AlicaContext.h

```diff
@@ -457,6 +458,9 @@ class AlicaContext
      * Get communication Handlers
      */
     AlicaCommunicationHandlers getCommunicationHandlers();
+
+    Blackboard& editBlackboard();
```

Comment:
I don't see a need for these methods & seem like they are out of place. Isn't it better to just get the blackboard & result store from the engine & pass to the solvers?

Reply:
understood, so it looks like a design problem like I mentioned here: https://github.com/rapyuta-robotics/alica/pull/303#discussion_r979680583

Reply:
can you add a comment here mentioning this circular dependency & add a TODO to fix it?